tests/test_transforms/test_embeddings/test_models/test_ts2vec.py

```diff
@@ -83,17 +83,53 @@ def test_encode_format(ts_with_exog_nan_begin_numpy, output_dims, segment_shape_
 
 
 def test_encode_pre_fitted(ts_with_exog_nan_begin_numpy, tmp_path):
-    model = TS2VecEmbeddingModel(input_dims=3, n_epochs=1)
-    assert model._is_fitted is False
+    model = TS2VecEmbeddingModel(input_dims=3)
+    model.fit(ts_with_exog_nan_begin_numpy, n_epochs=1)
+    path = pathlib.Path(tmp_path) / "tmp.zip"
+    model.save(path=path)
+
+    model_loaded = TS2VecEmbeddingModel.load(path=path)
+
+    np.testing.assert_array_equal(
+        model.encode_window(ts_with_exog_nan_begin_numpy), model_loaded.encode_window(ts_with_exog_nan_begin_numpy)
+    )
+    np.testing.assert_array_equal(
+        model.encode_segment(ts_with_exog_nan_begin_numpy), model_loaded.encode_segment(ts_with_exog_nan_begin_numpy)
+    )
+
+
+def test_not_freeze_fit(ts_with_exog_nan_begin_numpy, tmp_path):
+    model = TS2VecEmbeddingModel(input_dims=3)
+    model.fit(ts_with_exog_nan_begin_numpy, n_epochs=1)
+    model.freeze(is_freezed=False)
+    path = pathlib.Path(tmp_path) / "tmp.zip"
+    model.save(path=path)
```

Comment:
Should we really save model on disk? Coudn't we just make a `deepcopy`?

Reply:
I think the main way of manipulating of a pretrained model is saving and loading it
And we should be confident that during saving and loading all goes right